File: Source/JavaScript/model/interactions/InteractionArgument.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { BindingExpression } from '../common';

/**
 * One argument passed to an action.
 */
export interface InteractionArgument {
    name: string;
    value: BindingExpression;
}

export const InteractionArgumentPropertyNames: (keyof InteractionArgument)[] = ['name', 'value'];

File: Source/DotNET/Model/Interactions/InteractionArgument.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Scene.Model.Common;

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Represents one argument passed to an action.
/// </summary>
/// <param name="Name">The argument name on the target.</param>
/// <param name="Value">Where the value comes from.</param>
public record InteractionArgument(string Name, BindingExpression Value);
